### ISSUE TITLE
Create `Tab` and `Tabs` components

### ISSUE THREAD

**Issue by bmuenzenmeyer** [nodejs/nodejs.org#5817]
Create `Tab` and `Tabs` components using the new [figma design (direct link)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=423-27908&mode=dev). 

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).
- Review the [component creation guide](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#creating-components) to follow our standards.

---

### Details

![image](https://github.com/nodejs/nodejs.org/assets/298435/f601ac3a-4865-41d9-a2b9-32cf6865e4ee)

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Create a new directory called `Tab` within the [`Common` directory](https://github.com/nodejs/nodejs.org/tree/main/components/Common) that contains all the new code.
- Use the new tailwind color and typography system to assign properties. For example, the selected tab uses `Green/400` per the figma.
- Consider the [defined accessibility guidelines of a tab pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/). Meaning, tabs should 
- Add a storybook story file `components/Common/Tab/index.stories.tsx` which exercises each of the component's states.
  - Suggest a `Tabs` > `Tab`, `Tab`, `Tab` arrangement to showcase the functionality and interactivity of the components.  

Suggested `Tabs` props include:

- `onChange` (function): see discussion below

Suggested `Tab` props include: _(more might be required, especially for accessibility)_

- `text` (string):  (`Using a Package Manager` in the figma)
- `active` (boolean): shows the selected tab
- accept `children` content

There are 4 states to capture within styles and stories:

- Active, Light
- Not-Active, Light
- Active, Dark
- Not-Active, Dark

### Discussion

The API of `Tabs` and `Tab` can highly vary. MUI, for example, goes through great effort to only [define a single `onChange` handler on `Tabs`](https://mui.com/material-ui/react-tabs/) which then [gets cloned down to the enumerated `Tab` children](https://github.com/mui/material-ui/blob/69c035ef4f498ca666c375ab6b2ed87f57b16650/packages/mui-material/src/Tabs/Tabs.js#L676-L702). I know we could do something simpler, (and chasing the API surface of a project like MUI comes with it all kinds of other problems) but at the expense of a slightly uglier API surface requiring prop drilling. Perhaps that's okay to start with. I wonder how the @nodejs/nodejs-website teams feels about either approach, or others.

**Comment by bmuenzenmeyer** [comment]
@ovflowd this one has a bit more functionality associated with it. any feedback welcome before a contributor picks it up

**Comment by araujogui** [comment]
I can work on this component.

> The API of Tabs and Tab can highly vary. MUI, for example, goes through great effort to only [define a single onChange handler on Tabs](https://mui.com/material-ui/react-tabs/) which then [gets cloned down to the enumerated Tab children](https://github.com/mui/material-ui/blob/69c035ef4f498ca666c375ab6b2ed87f57b16650/packages/mui-material/src/Tabs/Tabs.js#L676-L702). I know we could do something simpler, (and chasing the API surface of a project like MUI comes with it all kinds of other problems) but at the expense of a slightly uglier API surface requiring prop drilling. Perhaps that's okay to start with. I wonder how the https://github.com/orgs/nodejs/teams/nodejs-website teams feels about either approach, or others.

I personally like the [Radix UI](https://www.radix-ui.com/primitives/docs/components/tabs) approach

**Comment by bmuenzenmeyer** [comment]
@araujogui I don't have super strong opinions about the methods. eager to hear what others think

**Comment by bmuenzenmeyer** [comment]
@Harkunwar since you are helping with other issue creation, pinging you too for a second look

**Comment by ovflowd** [comment]
> I can work on this component.
> 
> > The API of Tabs and Tab can highly vary. MUI, for example, goes through great effort to only [define a single onChange handler on Tabs](https://mui.com/material-ui/react-tabs/) which then [gets cloned down to the enumerated Tab children](https://github.com/mui/material-ui/blob/69c035ef4f498ca666c375ab6b2ed87f57b16650/packages/mui-material/src/Tabs/Tabs.js#L676-L702). I know we could do something simpler, (and chasing the API surface of a project like MUI comes with it all kinds of other problems) but at the expense of a slightly uglier API surface requiring prop drilling. Perhaps that's okay to start with. I wonder how the https://github.com/orgs/nodejs/teams/nodejs-website teams feels about either approach, or others.
> 
> I personally like the [Radix UI](https://www.radix-ui.com/primitives/docs/components/tabs) approach

We discussed in person (me @bmuenzenmeyer and @Harkunwar) and we are fond of RadixUI, we will update this issue later, but tl;dr we want to integrate Radix here.

**Comment by ovflowd** [comment]
@araujogui is this something you'd like to work on? Since we wanted to try out RadixUI?

**Comment by araujogui** [comment]
> @araujogui is this something you'd like to work on? Since we wanted to try out RadixUI?

Yeah, I'd be happy to

**Comment by ovflowd** [comment]
Also @araujogui when working on this don't forget to update COLLABORATOR_GUIDE.md adding that we use Radix UI and on the bottom "FAQ" section: "Why we use RadixUI?"

Worth mentioning:

- It is a minimalistic component library broken down in individual packages for each Component
- It already handles all `WAI-ARIA` and Accessibility shortcuts/bindings needed for Interactive Elements
- It allows us to focus on designing interactive Components without the effort of adding all the surrounding sugar and code needed to make the Component accessibility-friendly.

**Comment by AugustinMauroy** [comment]
Small question: why don't use Headless UI it's from tailwind ?

**Comment by ovflowd** [comment]
> Small question: why don't use Headless UI it's from tailwind ?

We debated in-person all the options. And we decided to use RadixUI. (During Open Source Summit EU)

There are multiple reasons:

1. RadixUI seems to provide way more Components
2. HeadlessUI does not support tree-shaking
3. RadixUI has one Component per NPM package and has 0 dependencies, meaning we only install what we need.
4. RadixUI is focused on Accessibility, HeadlessUI has _some_ accessibility features but not as many as Radix.
